### PR TITLE
Refactor View package

### LINKED ISSUE #75: Refactor View package
- Move getRoute() into a routable behaviour
- Make ViewAbstract class properties private
- Add getMimetype() method to ViewInterface
- Handling route escaping transparently

Breaking changes:

- The View class properties are now private. To migrate your code use the getter methods as defined in ViewInterface

### PR DESCRIPTION
**Pull request by johanjanssens**
resolve #75

### CODE REVIEW

**Comment by johanjanssens** [comment]
@ercanozkaya  Can you review please ? Question: do you think we should add a View::getVersion() method to make view versioning a first class citizen of a view ?

File: code/view/behavior/routable.php [hunk 1]
```diff
@@ -86,8 +101,15 @@ public function getRoute($route = '', $fqr = true)
         $escape = $this->getUrl()->isEscaped();
         $route  = $this->getObject('lib:dispatcher.router.route', array('escape' =>  $escape))->setQuery($parts);
 
-        //Add the host and the schema
-        if ($fqr === true)
+        //Determine of the url needs to be fully qualified
+        if($this->getMixer()->getFormat() == 'html') {
+            $fqe = is_bool($fqr) ? $fqr : false;
```
Comment (ercanozkaya) [inline code/view/behavior/routable.php]: @johanjanssens typo here. should be $fqr

File: code/view/json.php [hunk 2]
```diff
@@ -342,8 +326,10 @@ protected function _getEntityRoute(ModelEntityInterface $entity)
      */
     protected function _convertRelativeLinks(ViewContextInterface $context)
     {
-        if (is_array($this->_content) || $this->_content instanceof \Traversable) {
-            $this->_processLinks($this->_content);
+        $content = $context->content;
```
Comment (ercanozkaya) [inline code/view/json.php]: @johanjanssens you might wanna copy this as reference if $context->content can ever be a pure array.

**Comment by ercanozkaya** [comment]
@johanjanssens Left two notes, good to go other than that.

I wouldn't add versioning support directly to the view. It's only useful for public facing APIs and even in that case there are different takes whether or not to use versions.

File: code/view/json.php (hunk 2 above)
Comment (johanjanssens) [inline code/view/json.php]: Not sure I understand what you mean ?

Comment (ercanozkaya) [inline code/view/json.php]: this line will make a copy of $context->content if it's an array. In which case the result of processLinks won't be stored anywhere.

**Pull request merged**